starky/src/vanishing_poly.rs

```diff
@@ -83,3 +93,222 @@ pub(crate) fn eval_vanishing_poly_circuit<F, S, const D: usize>(
+            .collect::<Vec<_>>(),
+        z_last,
+        l_0,
+        l_last,
+    );
+
+    let vars = S::EvaluationFrame::from_values(
+        &local_values,
+        &next_values,
+        &public_inputs
+            .iter()
+            .copied()
+            .map(F::Extension::from_basefield)
+            .collect::<Vec<_>>(),
+    );
+
+    let lookup_vars = lookup_challenges.map(|l_c| LookupCheckVars {
+        local_values: auxiliary_polys.as_ref().unwrap()[..num_lookup_columns].to_vec(),
+        next_values: auxiliary_polys_next.as_ref().unwrap()[..num_lookup_columns].to_vec(),
+        challenges: l_c.to_vec(),
+    });
+
+    eval_vanishing_poly::<F, F::Extension, F::Extension, S, D, D>(
+        stark,
+        &vars,
+        &lookups,
+        lookup_vars,
+        ctl_vars,
+        &mut consumer,
+    );
+    consumer.accumulators()
+}
+
+pub(crate) fn eval_l_0_and_l_last_circuit<F: RichField + Extendable<D>, const D: usize>(
+    builder: &mut CircuitBuilder<F, D>,
+    n: ExtensionTarget<D>,
+    g: ExtensionTarget<D>,
+    x: ExtensionTarget<D>,
+    z_x: ExtensionTarget<D>,
+) -> (ExtensionTarget<D>, ExtensionTarget<D>) {
+    let one = builder.one_extension();
+    let l_0_deno = builder.mul_sub_extension(n, x, n);
+    let l_last_deno = builder.mul_sub_extension(g, x, one);
+    let l_last_deno = builder.mul_extension(n, l_last_deno);
+
+    (
+        builder.div_extension(z_x, l_0_deno),
+        builder.div_extension(z_x, l_last_deno),
+    )
+}
+
+/// Evaluates the constraints at a random extension point. It is used to bind the constraints.
+pub(crate) fn compute_eval_vanishing_poly_circuit<F, S, const D: usize>(
+    builder: &mut CircuitBuilder<F, D>,
+    stark: &S,
+    stark_opening_set: &StarkOpeningSetTarget<D>,
+    ctl_vars: Option<&[CtlCheckVarsTarget<F, D>]>,
+    lookup_challenges: Option<&Vec<Target>>,
+    public_inputs: &[Target],
+    alphas: Vec<Target>,
+    zeta: ExtensionTarget<D>,
+    degree_bits: usize,
+    degree_bits_target: Target,
+    num_lookup_columns: usize,
+) -> Vec<ExtensionTarget<D>>
+where
+    F: RichField + Extendable<D>,
+    S: Stark<F, D>,
+{
+    get_constraint_accumulators(
+        builder,
+        stark,
+        stark_opening_set,
+        ctl_vars,
+        lookup_challenges,
+        public_inputs,
+        alphas,
+        zeta,
+        degree_bits,
+        degree_bits_target,
+        num_lookup_columns,
+    )
+}
+
+/// Gets the constraint accumulators given a [`StarkOpeningSet`].
+pub(crate) fn get_constraint_accumulators<F, S, const D: usize>(
+    builder: &mut CircuitBuilder<F, D>,
+    stark: &S,
+    stark_opening_set: &StarkOpeningSetTarget<D>,
+    ctl_vars: Option<&[CtlCheckVarsTarget<F, D>]>,
+    lookup_challenges: Option<&Vec<Target>>,
+    public_inputs: &[Target],
+    alphas: Vec<Target>,
+    zeta: ExtensionTarget<D>,
+    degree_bits: usize,
+    degree_bits_target: Target,
+    num_lookup_columns: usize,
+) -> Vec<ExtensionTarget<D>>
+where
```

Comment:
nit: the first is just a wrapper of the second, so maybe best to keep only `compute_eval_vanishing_poly_circuit` so that we have `let vanishing_polys_zeta = compute_eval_vanishing_poly_circuit(` in the recursive verifier?